jitrebalance/test_jitrebalance.py

```diff
@@ -0,0 +1,78 @@
+from pyln.testing.fixtures import *  # noqa: F401, F403
+from pyln.testing.utils import wait_for
+from pprint import pprint
+import os
+import time
+
+
+plugin = os.path.join(os.getcwd(), 'jitrebalance.py')
+
+
+def test_simple_rebalance(node_factory):
+    """Simple rebalance that routes along a cycle to enable the original payment
+
+    l1 ---- l2 ---- l3 ----- l4
+              |    /
+              |   /
+              |  /
+               l5
+
+    We are going to drain the channel (l2, l3) of most of its funds and then
+    ask l1 to route through [l1, l2, l3, l4]. Under normal circumstances
+    that'd fail since (l2, l3) doesn't have sufficient funds. l2 however will
+    attempt to rebalance (l2,l3) using a circular route (l2, l5, l3, l2) to
+    get the required funds back.
+
+    """
+    print(plugin)
+    opts = [{}, {'plugin': plugin}, {}, {}, {}]
+    l1, l2, l3, l4, l5 = node_factory.get_nodes(5, opts=opts)
+    amt = 10**7
+
+    # Open the channels
+    channels = [(l1, l2), (l3, l2), (l3, l4), (l2, l5), (l5, l3)]
+    for src, dst in channels:
+        src.openchannel(dst, capacity=10**6)
+
+    # Drain (l2, l3) so that a larger payment fails later on
+    chan = l2.rpc.listpeers(l3.info['id'])['peers'][0]['channels'][0]
+    pprint(chan)
+    # Send 9 million millisatoshis + reserve + a tiny fee allowance from l3 to
+    # l2 for the actual payment
+    inv = l2.rpc.invoice(
```

Comment:
There is something I don't get with this test : shouldn't we drain __from__ `l2` ?

If I follow the docstring :
```
l2 however will attempt to rebalance (l2,l3) using a circular route (l2, l5, l3, l2) to get the required funds back.
```
then l2 would try to send a payment through `l3 => l2` while it's already exhausted ?

Reply:
We don't have capacity in the direction `(l2, l3)`, so we create a circular route `(l2,l5, l3, l2)` (opposite direction of what we are trying to rebalance). `(l2, l3)` is exhausted which is why we try the rebalancing, if `(l3, l2)` is also exhausted we can never rebalance correctly, so any attempt would fail.

Reply:
thanks